src/Deprecated/Engine/Choose/GroupingCollection.cs

```diff
@@ -448,7 +448,9 @@ IItemPropertyGrouping insertionPoint
                 ChangePropertyGroupCount(1);
             }
             else if (newGroup is Choose)
+            {
```

Comment:
Wondering if this one (and below) should be converted into switch statements? If not, would be nice to at least make them like `newGroup is BuildItemGroup big` so we can skip the cast below. No need to actually do that in this PR. Keeping things separate is good.

Reply:
They're both deprecated code. So once I sort out the change in #7192, it shouldn't run against the deprecated projects.

Reply:
Good point